easybuild/tools/docs.py

```diff
@@ -744,28 +744,65 @@ def list_toolchains_rst(tcs):
     """ Returns overview of all toolchains in rst format """
     title = "List of known toolchains"
 
-    # figure out column names
-    table_titles = ['name', 'compiler', 'MPI']
-    for tc in tcs.values():
-        table_titles.extend(tc.keys())
+    # Specify the column names for the table
+    table_titles = ['name', 'COMPILER', 'MPI', 'linalg', 'FFT']
 
+    # Set up column name : display name pairs
     col_names = {
-        'COMPILER_CUDA': 'CUDA compiler',
-        'SCALAPACK': 'ScaLAPACK',
+        'name': 'Name',
+        'compiler': 'Compiler(s)',
+        'linalg': "Linear algebra",
     }
 
-    table_titles = nub(table_titles)
+    # Create sorted list of toolchain names
+    sorted_tc_names = sorted(tcs.keys(), key=str.lower)
 
-    table_values = [[] for i in range(len(table_titles))]
-    table_values[0] = ['**%s**' % tcname for tcname in tcs.keys()]
+    # Create text placeholder to use for missing entries.
+    # Not good documentation style to italicize the parentheses enclosing
+    # italic text unless surrouning text is also italic.
+    none_txt = '(*none*)'
 
-    for idx in range(1, len(table_titles)):
-        for tc in tcs.values():
-            table_values[idx].append(', '.join(tc.get(table_titles[idx].upper(), [])))
+    # Initialize an empty list of lists for the table data
+    table_values = [[] for i in range(len(table_titles))]
 
+    for col_id, col_name in enumerate(table_titles):
+        if col_name == 'name':
+            # toolchain names column gets bold face entry
+            table_values[col_id] = ['**%s**' % tcname for tcname in sorted_tc_names]
+        else:
+            for tc_name in sorted_tc_names:
+                tc = tcs[tc_name]
+                # Cray is a special snowflake
+                if tc_name.find('Cray') >= 0:
+                    if col_name == 'COMPILER':
+                        entry = 'PrgEnv-cray'
```

Comment:
This won't always be `PrgEnv-cray`, only for `CrayCCE` this is correct.
You can grab the correct name via `tc` though:

```python
entry = tc[col_name.upper()]

Reply:
That returns a list from which the string must be extracted for printing.  If this is done for Cray the same way it is for the other toolchains, then that would be
```
entry = ', '.join(tc[col_name.upper()])
```
Is that how you would like it done?

Reply:
Yes, I overlooked the list part, sorry.